src/route/index.ts

```diff
@@ -1291,46 +1292,50 @@ export async function initApiServer(
         schema: {
           body: {
             type: "object",
-            required: ["xdr", "network_url", "network_passphrase"],
+            required: ["xdr", "network"],
             properties: {
               xdr: { type: "string" },
-              network_url: { type: "string" },
-              network_passphrase: { type: "string" },
+              network: { type: "string" },
             },
           },
         },
         handler: async (
           request: FastifyRequest<{
             Body: {
               xdr: string;
-              network_url: string;
-              network_passphrase: string;
+              network: NetworkNames;
             };
           }>,
           reply,
         ) => {
-          const { xdr, network_url, network_passphrase } = request.body;
+          const { xdr, network } = request.body;
 
           try {
-            const Sdk = getSdk(network_passphrase as Networks);
-            const tx = Sdk.TransactionBuilder.fromXDR(xdr, network_passphrase);
-            const server = new Sdk.SorobanRpc.Server(network_url);
+            const networkPassphrase = Networks[network];
+            if (!networkPassphrase) {
+              reply.code(400).send({ error: ERROR.UNSUPPORTED_NETWORK });
+            }
+            const Sdk = getSdk(networkPassphrase);
+            const tx = Sdk.TransactionBuilder.fromXDR(xdr, networkPassphrase);
+            const server = await getServer(network, stellarRpcConfig);
```

Comment:
got it, so this is the main fix which will set `allowHttp: true` when needed. Is my understanding correct?